rclcpp/src/rclcpp/context.cpp

```diff
@@ -23,25 +23,47 @@
 
 #include "rcl/init.h"
 #include "rcl/logging.h"
+
 #include "rclcpp/detail/utilities.hpp"
 #include "rclcpp/exceptions.hpp"
 #include "rclcpp/logging.hpp"
+
+#include "rcutils/error_handling.h"
+#include "rcutils/macros.h"
+
 #include "rmw/impl/cpp/demangle.hpp"
 
+#include "./logging_mutex.hpp"
+
 /// Mutex to protect initialized contexts.
 static std::mutex g_contexts_mutex;
 /// Weak list of context to be shutdown by the signal handler.
 static std::vector<std::weak_ptr<rclcpp::Context>> g_contexts;
 
 using rclcpp::Context;
 
+extern "C"
+{
 static
-std::shared_ptr<std::mutex>
-get_global_logging_configure_mutex()
+void
+rclcpp_logging_output_handler(
+  const rcutils_log_location_t * location,
+  int severity, const char * name, rcutils_time_point_value_t timestamp,
+  const char * format, va_list * args)
 {
-  static auto mutex = std::make_shared<std::mutex>();
-  return mutex;
+  std::shared_ptr<std::recursive_mutex> logging_mutex;
+  try {
+    logging_mutex = get_global_logging_mutex();
+  } catch (std::exception & ex) {
+    RCUTILS_SAFE_FWRITE_TO_STDERR(ex.what());
+    return;
+  }
```

Comment:
I think this needs a `catch (...)` statement too.

Reply:
Fixed in https://github.com/ros2/rclcpp/pull/1125/commits/b29cdd698b0b8778d036b73a55f49b976b37fca4.